app/src/main/java/com/infomaniak/drive/ui/addFiles/AddFileBottomSheetDialog.kt

```diff
@@ -113,17 +111,8 @@ class AddFileBottomSheetDialog : BottomSheetDialogFragment() {
             registerPermissions(this@AddFileBottomSheetDialog) { authorized -> if (authorized) openCamera() }
         }
 
-        uploadFilesHelper = UploadFilesHelper(
-            fragment = this@AddFileBottomSheetDialog,
-            onOpeningPicker = {
-                trackNewElement("uploadFile")
-                binding.documentUpload.isEnabled = false
-            },
-            onResult = { findNavController().popBackStack() },
```

Comment:
You replaced this instance with a different one in. the viewModel (activity instead of fragment; no onResult)
Won't it cause trouble ? 
Edit: I see now that you navigate to the UploadDialog, I guess this is why you don't want to popbackstack, because it pop out of the dialog ? 
The problem now is that if you back from the list, you will stay on the list